src/snowflake/connector/cache.py

```diff
@@ -58,6 +58,9 @@ def __init__(
         self._cache: dict[K, CacheEntry[V]] = {}
         self._lock = Lock()
         self._reset_telemetry()
+        # indicate whether the cache is modified or not, this variable is for
+        # SFDictFileCache to determine whether to dump cache to file when _save is called
+        self._cache_modified = False
```

Comment:
We should move all of these to `SFDictFileCache`, this class has no use for this information!

Reply:
yep, you're right!